omreader/src/Main.scala

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 Jiuyang Liu <liu@jiuyang.me>
+
+package org.chipsalliance.t1.omreader
+
+import java.io.BufferedInputStream
+import mainargs._
+import chisel3.panamaom._
+import org.chipsalliance.t1.omreaderlib._
+
+object Main {
+  implicit object PathRead extends TokensReader.Simple[os.Path] {
+    def shortName = "path"
+    def read(strs: Seq[String]): Either[String, os.Path] = Right(os.Path(strs.head, os.pwd))
+  }
+
+  @main
+  def run(
+    @arg(name = "mlirbc-file") mlirbcFile: Option[os.Path],
+    @arg(name = "class-name") className: String,
+    @arg(name = "dump-methods") dumpMethods: Flag,
+    @arg(name = "eval") eval: Option[String],
+  ) = {
+    val omReader = mlirbcFile match {
+      case Some(path) => OMReader.fromFile(path, className)
+      case None =>
+        val stdin = new BufferedInputStream(System.in)
+        val bytes = Stream.continually(stdin.read).takeWhile(_ != -1).map(_.toByte).toArray
+        OMReader.fromBytes(bytes, className)
+    }
+    val t1Reader = omReader.t1Reader
+
+    if (eval.nonEmpty) {
+      println(SimpleInputEval(omReader.entryUnderlying, eval.get))
+    } else if (dumpMethods.value) {
+      t1Reader.dumpMethods()
+    } else {
+      t1Reader.dumpAll()
+    }
+  }
+
+  def main(args: Array[String]): Unit = ParserForMethods(this).runOrExit(args)
+}
+
+object SimpleInputEval {
+  def apply(entry: PanamaCIRCTOMEvaluatorValue, input: String): PanamaCIRCTOMEvaluatorValue = {
+    input.split("\\.").foldLeft(entry) {
+      case (obj, field) =>
+        if (field.forall(_.isDigit)) {
+          obj.asInstanceOf[PanamaCIRCTOMEvaluatorValueList].getElement(field.toLong)
+        } else {
+          obj.asInstanceOf[PanamaCIRCTOMEvaluatorValueObject].field(field)
+        }
+    }
+  }
+}
```

Comment:
Don't do it.

Reply:
This is also for debugging purpose only, like `dump-` methods.